containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -314,33 +338,46 @@ def metadata_path(self):
         )
         return BucketPath(bucket=self.metadata_bucket, key=key)
 
+    def schema_path(self, table_name: str) -> BucketPath:
+        """
+        Path to the latest version schema file for a given table
+        """
+        key = os.path.join(
+            self.name,
+            self.latest_version,
+            table_name,
+            "schema.json",
+        )
+        return BucketPath(bucket=self.metadata_bucket, key=key)
+
     @staticmethod
-    def metadata_spec_prefix(bucket_name: str | None = None) -> BucketPath:
+    def specification_prefix(metadata_type: str, bucket_name: str | None = None) -> BucketPath:
```

Comment:
I don't think you need to accept a string here - you could type it to `MetadataType` and force the caller to pass in the correct value, instead of doing a lookup in the method. That way the type checker can catch the errors instead of it only being caught at runtime.